steps/measure.py

```diff
@@ -139,15 +152,53 @@ def hamiltonian_analysis(
     else:
         expecval = None
 
-    K_coeff, nterms, frame_meas = estimate_nmeas_for_operator(operator, decomposition_method, expecval)
-    save_nmeas_estimate(nmeas=K_coeff, nterms=nterms, frame_meas=frame_meas, filename='hamiltonian_analysis.json')
-   
+    K_coeff, nterms, frame_meas = estimate_nmeas_for_operator(
+        operator, decomposition_method, expecval
+    )
+    save_nmeas_estimate(
+        nmeas=K_coeff,
+        nterms=nterms,
+        frame_meas=frame_meas,
+        filename="hamiltonian_analysis.json",
+    )
+
+
+def grouped_hamiltonian_analysis(
+    groups: str, expectation_values: str = "None",
+):
+
```

Comment:
Please add a docstring :)

Reply:
Addressed in 0ff7e3c086d6027c44ee79a7c917c44d3a7fdb48